libvcx/src/messages/mod.rs

```diff
@@ -264,6 +264,13 @@ impl<'de> Deserialize<'de> for A2AMessageV2 {
         let value = Value::deserialize(deserializer).map_err(de::Error::custom)?;
         let message_type: MessageTypeV2 = serde_json::from_value(value["@type"].clone()).map_err(de::Error::custom)?;
 
+        let message_json = serde_json::ser::to_string(&value);
+        let message_type_json = serde_json::ser::to_string(&value["@type"].clone());
+
+        debug!("Deserializing A2AMessageV2 json: {:?}", &message_json);
```

Comment:
maybe trace? There's similar output on line 447 and it's trace. Also, it'd be better to embed `serde_json::ser::to_string(&value);` into the `debug!` macro, so we could avoid serialization if the value won't ever be used (if we've enabled `info` logs only).
Another option would be something like this
```
/// Determines if a message logged at the specified level in that module will
/// be logged.
///
/// This can be used to avoid expensive computation of log message arguments if
/// the message would be ignored anyway.
///
/// # Examples
///
/// ```edition2018
/// use log::Level::Debug;
/// use log::{debug, log_enabled};
///
/// # fn foo() {
/// if log_enabled!(Debug) {
///     let data = expensive_call();
///     debug!("expensive debug data: {} {}", data.x, data.y);
/// }
/// if log_enabled!(target: "Global", Debug) {
///    let data = expensive_call();
///    debug!(target: "Global", "expensive debug data: {} {}", data.x, data.y);
/// }
/// # }
/// # struct Data { x: u32, y: u32 }
/// # fn expensive_call() -> Data { Data { x: 0, y: 0 } }
/// # fn main() {}
/// ```
```